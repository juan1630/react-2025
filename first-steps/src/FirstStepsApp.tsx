import { ItemCounter } from "./ShoppingCart/ItemCounter";

interface ItemInCar {
  name: string
  quantity: number
}

const ProductsInCart: ItemInCar[] = [
  { name: 'Nintendo Switch 2' , quantity: 2 },
  { name: 'Pro controller', quantity: 10 },
  { name: 'Super Smash', quantity: 5 },
]
export function FirstStepsApp() {
  return (
    <>
    {
      ProductsInCart.map( ({ name, quantity }) => (<ItemCounter key={name} name={name}  quantity={quantity} />))
    }
    </>
  );
}
