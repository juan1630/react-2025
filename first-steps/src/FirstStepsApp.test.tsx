import { render, screen } from "@testing-library/react";
import { afterEach, describe, expect, test, vi } from "vitest";
import { FirstStepsApp } from './FirstStepsApp'


const mockItemCounter = vi.fn((props: unknown)=> {
    return <div data-testid="ItemCounter"/>
})

vi.mock('./ShoppingCart/ItemCounter', ()=>({
    ItemCounter: (props:unknown)=> mockItemCounter(props),
}))

// vi.mock('./ShoppingCart/ItemCounter', ()=>({
//     ItemCounter: (props:unknown)=> <div data-testid="ItemCounter" name={props.name} quantity={props.quantity} />,
// }))


describe('FirstStepsApp.tsx', ()=> {

    afterEach(()=>{
        vi.clearAllMocks() //limpia todos los mocks antes de cada prueba
    })

    test('Should match snpashot', ()=> {
        const {container} = render(<FirstStepsApp/>)
        expect(container).toMatchSnapshot()
    })

     test('Should render the correct number of itemCounter components', ()=> {
         render(<FirstStepsApp/>)
        
         const itemCounters = screen.getAllByTestId('ItemCounter') 
        expect(itemCounters.length).toBe(3)
        
    })

    test('Should render with correct counter', ()=> {
         render(<FirstStepsApp/>)


         expect(mockItemCounter).toHaveBeenCalledTimes(3)
         expect(mockItemCounter).toHaveBeenCalledWith({ name: 'Nintendo Switch 2' , quantity: 2 })
         expect(mockItemCounter).toHaveBeenCalledWith({ name: 'Pro controller', quantity: 10 })
         expect(mockItemCounter).toHaveBeenCalledWith({ name: 'Super Smash', quantity: 5 })
    })
})