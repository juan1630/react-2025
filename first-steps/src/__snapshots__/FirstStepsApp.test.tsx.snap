// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepsApp.tsx > Should match snpashot 1`] = `
<div>
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
</div>
`;
